src/lamp_py/ingestion/convert_gtfs_rt.py

```diff
@@ -353,38 +343,160 @@ def gz_to_pyarrow(
+            local_folder,
+            "new_table.parquet",
+        )
+        pq.write_table(table, new_table_path)
+
+        join_table_path = os.path.join(
+            local_folder,
+            "join_table.parquet",
+        )
+        join_ds = pd.dataset((local_path, new_table_path)).sort_by(
+            self.detail.table_sort_order
+        )
+        with pq.ParquetWriter(join_table_path, schema=table.schema) as writer:
+            for batch in join_ds.to_batches(batch_size=self.pq_batch_size):
+                writer.write_batch(batch)
+
+        os.replace(join_table_path, local_path)
+        os.remove(new_table_path)
 
-            sort_log = ProcessLogger(
-                "pyarrow_sort_by", table_rows=table.num_rows
+    def continuous_pq_update(self, table: pyarrow.Table) -> None:
+        """
+        Continuous updating of local parquet files that are synced with S3
+        """
+        log = ProcessLogger("continuous_pq_update")
+        log.log_start()
+        try:
+            partition_dt = self.partition_dt(table)
+
+            part_folder = os.path.join(
+                DEFAULT_S3_PREFIX,
+                str(self.config_type),
+                f"year={partition_dt.year}",
+                f"month={partition_dt.month}",
+                f"day={partition_dt.day}",
+                f"hour={partition_dt.hour}",
             )
-            sort_log.log_start()
-            if self.detail.table_sort_order is not None:
-                table = table.sort_by(self.detail.table_sort_order)
-            sort_log.log_complete()
-
-            write_parquet_file(
-                table=table,
-                file_type=s3_prefix,
-                s3_dir=os.path.join(
-                    os.environ["SPRINGBOARD_BUCKET"],
-                    DEFAULT_S3_PREFIX,
-                    s3_prefix,
-                ),
-                partition_cols=["year", "month", "day", "hour"],
-                visitor_func=self.send_metadata,
+            local_path = os.path.join(
+                self.tmp_folder,
+                part_folder,
+                f"{partition_dt.isoformat()}.parquet",
+            )
+
+            log.add_metadata(part_folder=part_folder, local_path=local_path)
+
+            # no existing table file, write new file
+            if self.sync_with_s3(local_path) is False:
+                pq.write_table(table, local_path)
+
+            # join existing table file with new table data
+            else:
+                self.merge_local_pq(table, local_path)
+
+            upload_path = local_path.replace(
+                self.tmp_folder, os.environ["SPRINGBOARD_BUCKET"]
             )
+            upload_file(local_path, upload_path)
+            self.send_metadata(upload_path)
 
-        except Exception:
+            log.log_complete()
+
+        except Exception as exception:
+            shutil.rmtree(
+                os.path.join(self.tmp_folder, part_folder), ignore_errors=True
+            )
```

Comment:
If the assertion in partition_dt throws, then this block will also throw since part folder is None.

Reply:
updated this to just nuke the whole `self.config_type` folder and start fresh